modules/frontend/journal/JournalFrontendModule.php

```diff
@@ -6,47 +6,55 @@ class JournalFrontendModule extends DynamicFrontendModule {
+			$iLimit = $aOptions[self::LIMIT_OPTION];
+		}
 		$aJournalParams = isset($aOptions['journal']) ? explode('-', $aOptions['journal']) : array();
 		// get journal ids
 		if(isset($aJournalParams[1])) {
 			$this->oJournalPage = PageQuery::create()->active()->findPk($aJournalParams[1]);
 			$mJournalId = explode(',', $this->oJournalPage->getPagePropertyValue('journal:journal_id', ''));
-
 		} else {
 			$mJournalId = $aJournalParams[0];
 		}
-
 		switch($sDisplayMode) {
-			case('recent_journal_entry_teaser') : return $this->renderRecentJournalEntryTeaser($mJournalId);
-			case('recent_comment_teaser') : return $this->renderRecentJournalCommentTeaser($mJournalId);
+			case('recent_journal_entry_teaser') : return $this->renderRecentJournalEntryTeasers($mJournalId, $iLimit);
+			case('recent_comment_teaser') : return $this->renderRecentJournalCommentTeasers($mJournalId, $iLimit);
 			case('recent_journal_entry') : return $this->renderRecentJournalEntry($mJournalId);
 			default: return null;
 		}
 	}
 
-	private function renderRecentJournalEntryTeaser($mJournalId) {
-		$oJournalEntry = FrontendJournalEntryQuery::create()->filterByJournalId($mJournalId)->mostRecentFirst()->findOne();
-		if($oJournalEntry === null) {
+	private function renderRecentJournalEntryTeasers($mJournalId, $iLimit) {
+		$oJournalEntries = FrontendJournalEntryQuery::create()->filterByJournalId($mJournalId)->mostRecentFirst()->limit($iLimit)->find();
+		if($oJournalEntries == null) {
 			return null;
 		}
-		$oTemplate = $this->constructTemplate('journal_entry_teaser');
-		$sHref = LinkUtil::link($oJournalEntry->getLink($this->oJournalPage));
-		$oTemplate->replaceIdentifier('title', TagWriter::quickTag('a', array('href' => $sHref), $oJournalEntry->getTitle()));
-		$oTemplate->replaceIdentifier('link_to_detail', $sHref);
-		// publish_at and date are deprecated as their usage varies between the template types
-		$oTemplate->replaceIdentifier('publish_at', $oJournalEntry->getPublishAt('U'));
-		$oTemplate->replaceIdentifier('date', LocaleUtil::localizeDate($oJournalEntry->getPublishAtTimestamp()));
-		$oTemplate->replaceIdentifier('date_timestamp', $oJournalEntry->getPublishAtTimestamp());
-		$oTemplate->replaceIdentifier('user_name', $oJournalEntry->getUserRelatedByCreatedBy()->getFullName());
-		$sTextShort = RichtextUtil::parseStorageForFrontendOutput($oJournalEntry->getTextShort());
-		$oTemplate->replaceIdentifier('text_short', $sTextShort);
-		$oTemplate->replaceIdentifier('text_short_truncated', StringUtil::truncate(strip_tags($sTextShort), 300));
-		return $oTemplate;
+		$oListTemplate = $this->constructTemplate('teaser_list');
+		foreach($oJournalEntries as $i => $oJournalEntry) {
+			$oTemplate = $this->constructTemplate('journal_entry_teaser');
+			$sHref = LinkUtil::link($oJournalEntry->getLink($this->oJournalPage));
+			$oTemplate->replaceIdentifier('title', TagWriter::quickTag('a', array('href' => $sHref), $oJournalEntry->getTitle()));
```

Comment:
We have lots of inconsistencies like the recently cleaned up date identifiers. We should also try to clean up title and link so that we have one of each that always consistently renders the link and the title and maybe one that renders a linked title only in teaser/list contexts and an unlinked title in detail contexts.

Reply:
Yes, that's very true, let's see ...